gpcontrib/arenadata_toolkit/sql/upgrade_test.sql

```diff
@@ -0,0 +1,155 @@
+
+-- Create, adb_create_tables, adb_collect_table_stats and alter extension:
+	CREATE EXTENSION arenadata_toolkit VERSION '1.0';
+	IF (from_version != '1.0')
+	THEN
+		EXECUTE FORMAT($fmt$ALTER EXTENSION arenadata_toolkit
+						UPDATE TO %1$I;$fmt$, from_version);
+	END IF;
+	PERFORM arenadata_toolkit.adb_create_tables();
+	PERFORM arenadata_toolkit.adb_collect_table_stats();
+	EXECUTE FORMAT($fmt$ALTER EXTENSION arenadata_toolkit
+						UPDATE TO %1$I;$fmt$, to_version);
+
+-- Check the result
+	IF to_version = (SELECT extversion
+				     FROM pg_extension
+				     WHERE extname='arenadata_toolkit')
+	THEN
+		RETURN NEXT 'alter, create_tables and collect_table_stats check';
+	END IF;
+
+-- Check field "tablespace_location" and table "db_files_history_backup_YYYYMMDDtHHMMSS"
+-- which were added at version 1.4
+	PERFORM arenadata_toolkit.adb_create_tables();
+
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='db_files_current' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT 'tablespace_location at arenadata_toolkit.db_files_current check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='__db_files_current' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT 'tablespace_location at arenadata_toolkit.__db_files_current check';
+	END IF;
+	IF EXISTS (SELECT 1
+				   FROM information_schema.columns
+				   WHERE table_schema='arenadata_toolkit' AND
+				         table_name='__db_files_current_unmapped' AND
+				         column_name='tablespace_location')
+	THEN
+		RETURN NEXT 'tablespace_location at arenadata_toolkit.__db_files_current_unmapped check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM information_schema.columns
+			   WHERE table_schema='arenadata_toolkit' AND
+			         table_name='db_files_history' AND
+			         column_name='tablespace_location')
+	THEN
+		RETURN NEXT 'tablespace_location at arenadata_toolkit.db_files_history check';
+	END IF;
+	IF EXISTS (SELECT 1
+			   FROM pg_tables
+			   WHERE schemaname='arenadata_toolkit' AND
+			         tablename like 'db_files_history_backup%')
+	THEN
+		RETURN NEXT 'db_files_history_backup check';
+	END IF;
+
+-- Cleanup before next step
+	DROP EXTENSION arenadata_toolkit;
+
+-- Check create extension with the latest version after current was installed and dropped
+	CREATE EXTENSION arenadata_toolkit;
+
+-- Check the result
+	IF EXISTS (SELECT 1
+			   FROM pg_available_extensions
+			   WHERE name='arenadata_toolkit' AND
+			         default_version=installed_version)
+	THEN
+		RETURN NEXT 'create the latest check';
+	END IF;
+
+-- Cleanup
+	DROP EXTENSION arenadata_toolkit;
+	DROP SCHEMA arenadata_toolkit CASCADE;
+
+END$$
+LANGUAGE plpgsql VOLATILE
+EXECUTE ON MASTER;
+
+SELECT do_upgrade_test_for_arenadata_toolkit('1.0', '1.1') ORDER BY 1;
```

Comment:
I think these tests are out of specification.

Reply:
I think another: 1-3 steps must check update between two adjacent versions of toolkit (for each pair). The last step must check installing toolkit at non-empty schema (if toolkit was installed and removed).

Reply:
> pre "extension era" arendata_toolkit schema exists

I think it means arenadata_toolkit from bundle as it was before it became an extension.

Reply:
How do tests with versions 1.0 -> 1.1 -> 1.2 -> 1.3 relate to the current patch to add a new column?

Reply:
> I think these tests should be added earlier

Still, I don’t quite understand what tests 1.0 -> 1.1 -> 1.2 -> 1.3 check?



> I try to imitate it by executing `adb_create_tables()` and `adb_collect_table_stats()` and dropping extension. Schema `areandata_toolkit` is not dropped and all tables exist.

OK, but specification says that we should call adb_create_tables and adb_collect_table_stats in this check.
Moreover, the bundle version of the extension contains objects with a different structure than those that remain after removing the extension.

Reply:
I thin we need tests for following
1) bundle version -> 1.4
2) 1.0 -> 1.4
3) 1.1 -> 1.4
4) 1.2 -> 1.4
5) 1.3 -> 1.4
with scenarios from specification.
Because the customers will have it.

Reply:
Is it possible to organize a cycle for updating from all previous versions to the current version, so as not to edit these tests in new versions?

Reply:
> Is it possible to organize a cycle for updating from all previous versions to the current version, so as not to edit these tests in new versions?

I know only how can we get first and last versions of extension (without intermediate versions):
```sql
postgres=# select version from pg_available_extension_versions where name = 'arenadata_toolkit';
 version 
---------
 1.4
 1.0
(2 rows)
```
But I little optimize the test, to do all at one query.